edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -210,10 +210,21 @@ async Task<Twin> GetTwinFunc()
             catch (Exception e)
             {
                 Events.ErrorGettingTwin(e);
+
+                if (!forceNewModuleClient && this.RetryWithNewModuleClient(e))
+                {
+                    return await this.GetTwinFromIoTHub(true);
```

Comment:
If this throws, will the exception bubble up? I beleive we don't want that.

Reply:
since this method already caught everything, it will not have any exception thrown from this method.  When exception happens, it will catch and return Option.None<Twin>().

Reply:
ah yes